app/templates/components/archive-project-milestone-list-item.hbs

```diff
@@ -53,11 +53,13 @@
       </small>
     {{else}}
       <br>
-      <small class="display-block">
-        <DateDisplay
-          @date={{this.milestone.dcpActualenddate}}
-        />
-      </small>
+      {{#if this.milestone.dcpActualenddate}}
+        <small class="display-block">
+          <DateDisplay
+            @date={{this.milestone.dcpActualenddate}}
+          />
+        </small>
+      {{/if}}
```

Comment:
We could also make the DateDisplay component no-op if the date property is empty or invalid